tools/parse-ramdisk-logs.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import re
+import sys
+
+# extraction datetime is always irrelevant
+EXTRACTION_TS = re.compile(r"^[0-9TZ\.:\-]* ?")
+CONTENTS = re.compile(r"\*{12} Contents of (?P<path>/.*)\.tar\.gz ")
+ENTRY = re.compile(r"\*{4} Entry: (?P<entry>.*) \*{4}")
+
+
+def log(msg):
+    print(msg, file=sys.stderr)
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument("source")
+parser.add_argument("destination")
+args = parser.parse_args()
+
+# Visitor is an object that is currently responsible for handling new lines.
+# These objects are hierarchical: top level is host (Ironic node), second level
+# is specific source (file or command). Initially, visitor is None: such lines
+# are skipped.
+visitor = None
+
+# Skipped is the counter for initially skipped lines.Unfortunately, this may
+# include not just the inotify chatter but also useful lines if the logs have
+# been rotated. There is nothing that can be done here: the file name is only
+# available in the beginning of a dump.
+skipped = 0
+
+
+class VisitorEntry:
+    def __init__(self, dest, entry):
+        assert entry is not None
+        path = os.path.join(dest, entry)
+        os.makedirs(os.path.dirname(path), exist_ok=True)
+        self.file = open(path, "wt")
```

Comment:
```suggestion
        self.file = open(path, "wt", encoding="utf-8")
```
I'm assuming here the log files are utf-8, adjust as needed.

Reply:
I agree about reading files, but IMO we should use the default encoding when writing the result (which in most practical cases is still utf-8).